ta/pkcs11/src/processing_rsa.c

```diff
@@ -0,0 +1,544 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2018-2020, Linaro Limited
+ */
+
+#include <assert.h>
+#include <pkcs11_ta.h>
+#include <tee_api_defines.h>
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+
+#include "attributes.h"
+#include "object.h"
+#include "pkcs11_token.h"
+#include "processing.h"
+
+enum pkcs11_rc
+pkcs2tee_proc_params_rsa_pss(struct active_processing *proc,
+			     struct pkcs11_attribute_head *proc_params)
+{
+	struct serialargs args = { };
+	enum pkcs11_rc rc = PKCS11_CKR_GENERAL_ERROR;
+	struct rsa_pss_processing_ctx *ctx = NULL;
+	uint32_t hash = 0;
+	uint32_t mgf = 0;
+	uint32_t salt_len = 0;
+
+	serialargs_init(&args, proc_params->data, proc_params->size);
+
+	rc = serialargs_get(&args, &hash, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	rc = serialargs_get(&args, &mgf, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	rc = serialargs_get(&args, &salt_len, sizeof(uint32_t));
+	if (rc)
+		return rc;
+
+	if (serialargs_remaining_bytes(&args))
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	proc->extra_ctx = TEE_Malloc(sizeof(struct rsa_pss_processing_ctx),
+				     TEE_USER_MEM_HINT_NO_FILL_ZERO);
+	if (!proc->extra_ctx)
+		return PKCS11_CKR_DEVICE_MEMORY;
+
+	ctx = proc->extra_ctx;
+
+	ctx->hash_alg = hash;
+	ctx->mgf_type = mgf;
+	ctx->salt_len = salt_len;
+
+	return PKCS11_CKR_OK;
+}
+
+enum pkcs11_rc pkcs2tee_algo_rsa_pss(uint32_t *tee_id,
```

Comment:
I think `pkcs2tee_algo_rsa_pss()` (as well as `pkcs2tee_algo_ec()` in _processing_ec.c_, i'll create a PR), deserves an inline description:
```
/*
 * Check or set TEE algorithm identifier upon PKCS11 mechanism parameters 
 * @tee_id: Input and/or output TEE algorithm identifier
 * @proc_params: PKCS11 processing parameters
 */
```
I think we should clarify that `tee_id` can be a input data or an output data. Input when mechanism params are not needed to select the appropriate TEE Id, output when there are needed.

Reply:
OK